simularium/nbsv
# feature/custom-checkbox

Time Estimate or Size
=======
_Small 15 minutes_
_If you want to preview how the highlighting functionality will get integrated there is a [draft PR ](https://github.com/simularium/nbsv/pull/30)pointing at this branch._

Problem
=======
We have two types of checkboxes: regular antd checkbox for hiding agents and a star for highlighting

Solution
========
This custom component handles the rendering of the two types of checkboxes.
The implementing components are more concise and pass down minimal props: checkbox type, status, and onclick handler.

I am rendering a "highlight" star to show that the custom checkbox rendering functions work. 
-Clicking it has no effect
- Its status is simply the inverse of the other checkboxes state. 

CSS related styling out of scope, this is the code side of rendering the checkboxes.

* New feature (non-breaking change which adds functionality)

Steps to Verify:
----------------
1. Click the hiding checkbox and observe the two checkboxes, and their tooltip text.

Screenshots (optional):
-----------------------
![Screenshot 2024-03-21 at 3 26 25 PM](https://github.com/simularium/nbsv/assets/24981838/f67237f6-0edd-469f-9051-5ec9ac9dfea9)
